### ISSUE TITLE
Extend FeatureRanking interface for regression tasks

### ISSUE THREAD

**Issue by sabbatinif** [haifengl/smile#684]
It may be useful to have a feature ranking procedure applicable not only to classification tasks (e.g. `SignalNoiseRatio` and `SumSquaresRatio`, implementing the `FeatureRanking` interface), but also to regression tasks.
At the moment the `FeatureRanking` interface only accepts integer target vectors for calculating the feature rank.

**Comment by haifengl** [comment]
What feature selection criteria for regression are of interest?

**Comment by sabbatinif** [comment]
I have no strong preferences about the criteria. I can suggest something similar to [Python SciKit-Learn's `feature_selection.f_regression`](https://scikit-learn.org/stable/modules/generated/sklearn.feature_selection.f_regression.html). It consists of a sequential algorithm aimed at iteratively and greedly selecting the most relevant features of a dataset. It starts by training a temporary regressor on a single feature (the most correlated with respect to the output values) and it keeps repeating this operation by adding one feature at a time, always peaking the one that mostly increases the temporary regressor predictive performance. At the end of this process, features are ranked on the basis of their relevance. But any other criteria is useful for me

**Comment by kklioss** [comment]
We add `smile.feature.selection.FRegression` as a feature selection metric for regression. For more advanced method, please try `smile.feature.select.GAFE` that leverage genetic algorithm to select a subset of features for your choice of regression algorithm. You may also use `RandomForest` as a feature selection method as it reports `importance` of features based on their impact in the regression trees.
